controllers/gateway/apirule_controller.go

```diff
@@ -285,6 +269,50 @@ func handleDependenciesError(name string, err error) controllers.Status {
 	}
 }
 
+func discoverGateway(client client.Client, ctx context.Context, l logr.Logger, rule *gatewayv2alpha1.APIRule) (*networkingv1beta1.Gateway, bool, error) {
+	if rule.Spec.Gateway == nil {
+		return nil, false, fmt.Errorf("expected Gateway to be set")
+	}
+
+	// The regex pattern is the exact same as the one used in the APIRule validation
+	// Unfortunately usage of a constant is not possible here, as the Kubebuilder CRD validation interface is comment based
+	match, err := regexp.MatchString(`^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?/([a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)$`, *rule.Spec.Gateway)
```

Comment:
The original issue is that APIRule `spec.Gateway` is empty, correct? I would not duplicate regex matching here, but just check for the case when empty spec is produced (e.g. `spec.Gateway` is empty) on conversion, because of not-supported case.

Reply:
I found another bug unfortunately, you can edit the gateway format in v1beta1 CRD version.

This snippet assumes that the format is always OK, meaning that array out of bound error will happen on such occasion.
```go
	gatewayName := strings.Split(*rule.Spec.Gateway, "/")
 	gatewayNN := types.NamespacedName{
 		Namespace: gatewayName[0],
 		Name:      gatewayName[1],
 	}
```

Reply:
okay, so we can not then skip on matching again, perhaps just add comment (to not forget) to remove the double reg-ex definition when re get rid of v1beta1 and conversion.